### ISSUE TITLE
Anyone using provider in production/open-source apps?

### ISSUE THREAD

**Issue by sawankumarbundelkhandi** [rrousselGit/provider#62]
Similar to:
reactiveui/ReactiveUI#979
PrismLibrary/Prism#1016
brianegan/flutter_redux#86
felangel/bloc#139

If anyone is using the library then please give your input.

Use the following template:

If available on stores
iOS:
Android:

If open source
Source Code:

Your opinion
Version: ?
Years of experience: ?
Good: ?
Bad: ?

Thank you very much!

**Comment by HaithamSheshtawy** [comment]
I recommended this Structure for Provider.
https://medium.com/flutter-community/flutter-architecture-provider-implementation-guide-d33133a9a4e8

**Comment by amondnet** [comment]
[Diet Friends (다이어트 프렌즈)](https://apps.apple.com/app/id1466899980) is a chatbot app to help your diet. Currently only Korean is supported.

iOS: https://apps.apple.com/app/id1466899980
Android: not published yet

**Comment by Vadaski** [comment]
我也写了一篇 Provider 的中文介绍 https://juejin.im/post/5d00a84fe51d455a2f22023f

**Comment by phoenixsky** [comment]
Fun Android  with Provider mvvm
Source Code: [https://github.com/phoenixsky/fun_android_flutter/blob/master/README-EN.md](https://github.com/phoenixsky/fun_android_flutter/blob/master/README-EN.md)

```dart
Widget build(BuildContext context) {
    super.build(context);
    return ProviderWidget<StructureListModel>(
      model: StructureListModel(widget.cid),
      onModelReady: (model) => model.initData(),
      builder: (context, model, child) {
        if (model.busy) {
          return ViewStateSkeletonList(
            builder: (context, index) => ArticleSkeletonItem(),
          );
        } else if (model.error) {
          return ViewStateWidget(onPressed: model.initData);
        } else if (model.empty) {
          return ViewStateEmptyWidget(onPressed: model.initData);
        }
        return SmartRefresher(
            controller: model.refreshController,
            header: WaterDropHeader(),
            footer: RefresherFooter(),
            onRefresh: model.refresh,
            onLoading: model.loadMore,
            enablePullUp: true,
            child: ListView.builder(
                itemCount: model.list.length,
                itemBuilder: (context, index) {
                  Article item = model.list[index];
                  return ArticleItemWidget(item);
                }));
      },
    );
  }
```


Thanks `HaithamSheshtawy`  [provider-guide](https://medium.com/flutter-community/flutter-architecture-provider-implementation-guide-d33133a9a4e8)

**Comment by irperera** [comment]
can you use provider for authentication?

**Comment by guilhermeagostinelli** [comment]
> can you use provider for authentication?

Yes @irperera ! It works specially well with [firebase_auth](https://pub.dev/packages/firebase_auth), take a look [here](https://medium.com/flutter-community/flutter-firebase-login-using-provider-package-54ee4e5083c7) for an example of how you could do it.

**Comment by chnkrydv** [comment]
Yes, I use it in production.

Android app link - https://play.google.com/store/apps/details?id=in.ckzero.batua

My `main.dart` file looks like this, where all the stores (ChangeNotifierPrrovider extended classes) are fed to MyApp (UI) as MultiProviders:
`class AppWithStores extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MultiProvider(
      providers: [
        ChangeNotifierProvider<ModalStore>(create: (context) => AuthStore()),
        ChangeNotifierProvider<SettingsStore>(create: (context) => SettingsStore()),
        ChangeNotifierProvider<ExpensesStore>(create: (context) => ExpensesStore()),
        ChangeNotifierProvider<ExpenseEditStore>(create: (context) => ExpenseEditStore()),
        ChangeNotifierProvider<ChartsStore>(create: (context) => ChartsStore()),
      ],
      child: MaterialApp(
        debugShowCheckedModeBanner: false,
        title: 'batua',
        theme: ThemeData.light(),
        home: MyApp(),
      ),
    );
  }
}

void main() {
  SystemChrome.setSystemUIOverlayStyle(
      SystemUiOverlayStyle(statusBarBrightness: Brightness.light));

  runApp(new AppWithStores());
}`

**Comment by chnkrydv** [comment]
Yes, I use it in production.
Android app link (to see the scale of app) - https://play.google.com/store/apps/details?id=in.ckzero.batua

My `main.dart` looks like this. All the stores which are basically classes extending ChangeNotifier are provided as a club of providers within a MultiProvoder.
![image](https://user-images.githubusercontent.com/16503745/83402460-68aa7180-a424-11ea-91cd-026f0eb9f95b.png)

**Comment by pd4d10** [comment]
`provider` is heavily used in GitTouch, an open-source mobile app for Git services including GitHub, GitLab, Bitbucket and Gitea

iOS: https://apps.apple.com/us/app/gittouch/id1452042346
Android: https://play.google.com/store/apps/details?id=io.github.pd4d10.gittouch

Source Code: https://github.com/git-touch/git-touch

**Comment by tsinis** [comment]
Hey! I'm using Provider as a core of App Icon Tools (app for preview (shapes/animations), issue warnings, and generation of (launcher/desktop) app icons for Flutter apps (web, macOS, Windows, Android Adaptive, etc. icons including), which is available on all operating systems that Flutter offers to build.

iOS: coming not so soon
Android: coming soon, available in Releases
Web: https://app-icon.tools
macOS: coming not so soon, available in Releases
Linux: coming soon, available in Releases
Windows: coming soon, available in Releases

Source Code: https://github.com/tsinis/app_icon_tools

Version: 4.3.2+3

**Comment by nikhilbadyal** [comment]
More 
Notes Application. 
Source Code - https://github.com/nikhilbadyal/keep-notes
